code/game/objects/items/weapons/RCD.dm

```diff
@@ -104,10 +105,14 @@ obj/item/weapon/construction
 	if(!(A in view(7, get_turf(user))))
 		to_chat(user, "<span class='warning'>The \'Out of Range\' light on the [src] blinks red.</span>")
 		return FALSE
+	else
+		return TRUE
 
 /obj/item/weapon/construction/proc/prox_check(proximity)
-	if(!proximity)
-		return
+	if(proximity)
+		return TRUE
+	else
+		return FALSE
```

Comment:
I am returning TRUE because its a TRUE or FALSE statement for it to actually be valid or not.